src/TopStories/index.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { Component } from 'react';
+import { string, object, func, arrayOf, shape } from 'prop-types';
+import QuerySyntax from '../QuerySyntax/index.jsx';
+import queryBuilder from '../query-builder.js';  // eslint-disable-line
+import Story from './Story.jsx';
+import SortSelect from './SortSelect.jsx';
+import { fields } from '../fields';
+
+export default class TopStories extends Component {
+  static propTypes = {
+    stories: arrayOf(object).isRequired,
+    query: shape({
+      text: string,
+      date: object,
+    }),
+    onSortChange: func.isRequired,
+  }
+
+  state = {
+    showQuery: false,
+    sortType: (typeof this.props.query.sort === 'undefined' || this.props.query.sort === 'relevance') ? 'relevance' : 'date'
+  }
+
+  onShowQuery = () => {
+    this.setState({ showQuery: true });
+  }
+
+  onShowResults = () => {
+    this.setState({ showQuery: false });
+  }
+
+  onChangeSort = (e) => {
+    this.setState({ sortType: e.target.value });
+    let sortVal = e.target.value;
+    let newQuery = Object.assign({}, this.props.query, {
+      sort: sortVal
+    });
+    this.props.onSortChange(newQuery);
+  }
+
+  render() {
+    const { stories, query } = this.props;
+
+    return (
+      <div>
+        {
+          !this.state.showQuery
+            ? (
+                <div className="top-stories widget">
+                  <div className="widget--header">
+                    <h2 className="base--h2 widget--header-title">
+                      Top Stories
+                    </h2>
+                    <div className="widget--header-spacer" />
+                    <button
+                      className="base--button widget--header-button"
+                      href="#"
+                      onClick={this.onShowQuery}
+                    >
+                      View Query
+                    </button>
+                  </div>
+                  <p className="base--p top-stories--description">
+                    Find the most recent and relevant news articles.
+                  </p>
+                  <div className="sort-option">
+                    <span id="sort-label">
+                      Sort by:
+                    </span>
+                    <span>
+                      <SortSelect
+                        onChange={this.onChangeSort}
+                        currSelected={this.state.sortType}
+                      />
+                    </span>
+                  </div>
+                  <div className="top-stories--list">
+                    {
+                      stories.map(item =>
```

Comment:
sometimes I see `(arg) => {}` and other times I see `arg =>`

You got any of them linters?
![](https://imgflip.com/s/meme/Yall-Got-Any-More-Of.jpg)